src/doc/src/reference/environment-variables.md

```diff
@@ -260,9 +260,10 @@ corresponding environment variable is set to the empty string, `""`.
 * `CARGO_PRIMARY_PACKAGE` --- This environment variable will be set if the
   package being built is primary. Primary packages are the ones the user
   selected on the command-line, either with `-p` flags or the defaults based
-  on the current directory and the default workspace members. This environment
-  variable will not be set when building dependencies. This is only set when
-  compiling the package (not when running binaries or tests).
+  on the current directory and the default workspace members.
+  This variable will not be set when building dependencies,
+  unless a dependency is also a workspace member.
```

Comment:
I'm concerned that this statement isn't entirely true. In many (most?) cases, it won't be set for workspace member dependencies. For example in cargo's own workspace, it won't be set for any of its dependencies when doing `cargo run`.

I think to make this more accurate, perhaps change it to something like this?

```suggestion
  unless a dependency is also a workspace member that was also selected on the command-line.
```

However, as noted in the issue, what is "selected on the command-line" is sometimes not intuitive.

Reply:
I just want to highlight that there is a bug in how cargo cache's this behavior, which is described in #14678. I don't think that needs to change anything here, but indicates that the answer to when it is set is "buggy".

Reply:
Thank you. That is more accurate. Updated!

I think one of the problems here is that the paragraph wasn't well arranged. There is no paragraph or newline there. We may really need to give each variable a heading.